src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,78 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { ChangeEvent } from 'react';
+import { FilterTypes } from '../../types/Filter';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  filterBy: string
+  setFilterBy: (str: FilterTypes) => void,
+  query: string
+  setQuery: (str: string) => void,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({
+  filterBy,
+  setFilterBy,
+  query,
+  setQuery,
+}) => {
+  const handleSelect = (event: ChangeEvent<HTMLSelectElement>) => {
+    const status = event.target.value;
+
+    switch (status) {
+      case 'active':
+        setFilterBy(FilterTypes.Active);
+        break;
+      case 'completed':
+        setFilterBy(FilterTypes.Completed);
+        break;
+      case 'all':
+      default:
+        setFilterBy(FilterTypes.All);
+    }
+  };
```

Comment:
Maybe it can help 
```suggestion
  const handleSelect = (event: ChangeEvent<HTMLSelectElement>) => {
    const status = event.target.value as FilterTypes;
    
    setFilterBy(status);
  };
```